remediation/code-review/03-concat.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * Given two arrays, returns a new array that is a concatenation of the
+ * two given arrays.
+ *
+ * "Concatenation" means the two arrays are joined or glued together, end-to-end.
+ *
+ * JavaScript array's have a built-in concatenation method. Don't use it!
+ * The goal is to better understand how concatenation works, not navigate
+ * the JavaScript documentation.
+ *
+ * See https://en.wikipedia.org/wiki/Concatenation
+ *
+ * @example
+ * concat([1, 2, 3], [4, 5, 6]); // => [1, 2, 3, 4, 5, 6]
+ * concat([-10, undefined], [true, 'waffles']); // => [-10, undefined, true, 'waffles']
+ * concat([], []); // => []
+ * concat([20, 104], []); // => [20, 104]
+ * concat([], ['hello', 'world']); // => ['hello', 'world']
+ *
+ * @param {object[]} leftArray - The left array to concatenate
+ * @param {object[]} rightArray - The right array to concatenate
+ * @returns {object[]} The concatenation of the two given arrays
+ */
+
+ let dimes = [10,10];
+ let nickels = [5,5];
+ let dimeNickel = ['10','10','5','5'];
+
+function concat(leftArray, rightArray) {
+  let left = leftArray.join(",");
+  let right = rightArray.join(",");
+  let together = "";
+  together = left + "," + right;
+  console.log(together);
+  let concat = together.split(",");
```

Comment:
While this appears to work for some cases, its a shaky solution. What about arrays that have strings? The strings could have commas in them. How would this affect the result from `split(',')`.

In addition to this, there are other problems. If you have an array of numbers, they'll all be strings afterwards! That could lead to some unintended consequences.

It might help to try to think of the task in different terms. "Gluing two arrays together end-to-end" is one way of saying it. Another is "creating a new array that contains each individual item in two arrays while preserving order of the individual arrays". The prompt for this exercises asks you to not use `concat`, but there are other array methods. Can you find any in the docs that might help with adding individual items?